lib/puppet/provider/package.rb

```diff
@@ -53,4 +53,18 @@ def join_options(options)
       end
     end.flatten
   end
+
+  # Puppet on Windows prepends its paths to PATH.
+  # Provides a which method that does not traverse those paths.
+  # @api public
+  # @param command [String] the name of the executable to find.
+  # @return [String] the absolute path to the found executable.
+  def self.which_without_puppet_paths(command)
+    raise "This method is specific to Windows" unless Puppet::Util::Platform.windows?
```

Comment:
This is a bit scary to see - I would prefer that the check for Windows would simply return the command or a which output on non-Windows and do the following with Windows.